File: Week-2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    int index = (int) round(0.0588 * (((float) letters / (float) words) * 100) -
                            0.296 * (((float) sentences / (float) words) * 100) - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }

    return 0;
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; text[i]; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    if (text[0] == 0)
    {
        return 0;
    }
    for (int i = 0; text[i]; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    if (text[0] == 0)
    {
        return 0;
    }
    int sentences = 0;
    for (int i = 0; text[i]; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    if (sentences == 0)
    {
        sentences++;
    }
    return sentences;
}

File: Week-6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    index = 0.0588 * ((letters / words) * 100) - 0.296 * ((sentences / words) * 100) - 15.8

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(index)}")


def count_letters(text):
    letters = 0
    for c in text:
        if c.isalpha():
            letters += 1
    return letters


def count_words(text):
    words = 1
    if text[0] == " ":
        return 0

    for c in text:
        if c == " ":
            words += 1

    return words


def count_sentences(text):
    if text[0] == "":
        return 0

    sentences = 0
    for c in text:
        if c == "." or c == "!" or c == "?":
            sentences += 1
    if sentences == 0:
        sentences += 1

    return sentences


main()
